homeassistant/components/minecraft_server/__init__.py

```diff
@@ -78,7 +74,48 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
         # Migrate entities.
         await er.async_migrate_entries(hass, config_entry_id, _migrate_entity_unique_id)
 
-    _LOGGER.debug("Migration to version %s successful", config_entry.version)
+        _LOGGER.debug("Migration to version 2 successful")
+
+    # 2 --> 3: Use address instead of host and port in config entry.
+    if config_entry.version == 2:
+        _LOGGER.debug("Migrating from version 2")
+
+        config_data = config_entry.data
+
+        # Migrate config entry.
+        try:
+            address = config_data[CONF_HOST]
+            JavaServer.lookup(address)
+            host_only_lookup_success = True
+        except ValueError:
+            host_only_lookup_success = False
+
+        if not host_only_lookup_success:
```

Comment:
```suggestion
        except ValueError:
```

Maybe we should log also this error on debug?